mesonbuild/linkers/linkers.py

```diff
@@ -809,10 +811,14 @@ def get_asneeded_args(self) -> T.List[str]:
         return self._apply_prefix('-dead_strip_dylibs')
 
     def get_allow_undefined_args(self) -> T.List[str]:
-        return self._apply_prefix('-undefined,dynamic_lookup')
+        # iOS doesn't allow undefined symbols when linking
+        if self.system == 'ios':
+            return []
+        else:
+            return self._apply_prefix('-undefined,dynamic_lookup')
 
     def get_std_shared_module_args(self, target: 'BuildTarget') -> T.List[str]:
-        return ['-dynamiclib'] + self._apply_prefix('-undefined,dynamic_lookup')
+        return ["-dynamiclib"] + self.get_allow_undefined_args()
```

Comment:
The change from single to double quotes looks unrelated (and we generally prefer single quotes)